src/redux/books/books.js

```diff
@@ -37,8 +15,8 @@ export default function reducer(state = initialState, action = {}) {
     case LOAD_BOOKS:
       return books;
     case ADD_BOOK:
-      localStorage.setItem(JSON.stringify({ ...state, book }));
-      return { ...state, book };
+      localStorage.setItem('books', JSON.stringify([...state, book]));
+      return [...state, book];
     case EDIT_BOOK:
       return ([...state].map((book) => (action.id === book.id)) ? action.book : book);
     case REMOVE_BOOK:
```

Comment:
- [ ] Although this is working, you don't need to destructure the `state` variable to use its copy because the `map` function like the `filter` function, `concat` function returns a copy of the original array and does not mutate it. You can learn more about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

You would have something like eg: 
```
return state.map((book) => (action.id === book.id)) ? action.book : book;
```
Please fix it everywhere in your reducer function

Reply:
You are very correct.

Thanks for spotting that.